pkg/gcrcleaner/server.go

```diff
@@ -222,65 +222,67 @@ func (s *Server) clean(ctx context.Context, r io.ReadCloser) (map[string][]strin
+        )
+      )
+      WHEN "run.googleapis.com/Service" THEN JSON_QUERY_ARRAY(
+        resource.data,'$.spec.template.spec.containers'
+      )
+      WHEN "run.googleapis.com/Job" THEN JSON_QUERY_ARRAY(
+        resource.data,'$.spec.template.spec.template.spec.containers'
+      )
+    END
+    AS containers
+  FROM path_to_prod_metrics.running_images
+  WHERE readTime >= TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 7 day)
+), UNNEST(containers) AS container;`
+
+	bigQueryClient, err := bigquery.NewClient(ctx, projectID)
 	if err != nil {
-		return nil, http.StatusInternalServerError, fmt.Errorf("failed to create new asset client: %w", err)
+		return nil, http.StatusInternalServerError, fmt.Errorf("failed to create new BigQuery client: %w", err)
 	}
-	podFilter := NewAssetPodFilter(repos)
-	// gcloud beta asset list --organization=$ORGANIZATION_ID --asset-types='k8s.io/Pod' --content-type='resource' --format="value(resource.data.spec.containers.image)"
-	it := assetClient.ListAssets(ctx, &assetpb.ListAssetsRequest{
-		Parent:      fmt.Sprintf("organizations/%s", organizationId),
-		AssetTypes:  []string{"k8s.io/Pod"},
-		ContentType: assetpb.ContentType_RESOURCE,
-		PageSize:    1000,
-	})
-	for true {
-		asset, err := it.Next()
+
+	query := bigQueryClient.Query(recentlySeenImagesQuery)
+	query.Location = "US" // Location must match that of the dataset(s) referenced in the query.
+	queryIterator, err := query.Read(ctx)
+	if err != nil {
+		return nil, http.StatusInternalServerError, fmt.Errorf("failed to get query results from BigQuery: %w", err)
+	}
+
+	for {
+		var values []bigquery.Value
+		err := queryIterator.Next(&values)
 		if err == iterator.Done {
 			break
 		}
 		if err != nil {
-			return nil, http.StatusInternalServerError, fmt.Errorf("failed to list pods via cloudasset service: %w", err)
+			return nil, http.StatusInternalServerError, fmt.Errorf("failed to read row from BigQuery: %w", err)
 		}
-		podData := asset.Resource.Data
-		containers := podData.Fields["spec"].GetStructValue().Fields["containers"].GetListValue()
-		for _, container := range containers.Values {
-			image := container.GetStructValue().Fields["image"].GetStringValue()
-			s.logger.Debug("getting GKE pod images", "image", image)
-			err := podFilter.Add(image)
-			if err != nil {
-				return nil, http.StatusInternalServerError, fmt.Errorf("failed to parse container image: %w", err)
-			}
-		}
-	}
-	// gcloud beta asset list --organization=$ORGANIZATION_ID --asset-types='run.googleapis.com/Service' --content-type='resource' --format="value(resource.data.spec.template.spec.containers.image)"
-	it = assetClient.ListAssets(ctx, &assetpb.ListAssetsRequest{
-		Parent:      fmt.Sprintf("organizations/%s", organizationId),
-		AssetTypes:  []string{"run.googleapis.com/Service"},
-		ContentType: assetpb.ContentType_RESOURCE,
-		PageSize:    1000,
-	})
-	for true {
-		asset, err := it.Next()
-		if err == iterator.Done {
-			break
+		image, ok := values[0].(string)
```

Comment:
nits:
- is there a case in which `values[0]` is empty but `values[1]` isn't which could fail to throw err from `Next` but still cause this to error out?
- it doesn't look like we're using the second column of the query result, remove and leave this as is?

Reply:
IIUC, the iterator returns a row, which is an array of values (columns). Since there's only one column returned by this query--`image`--we just pick out the single item from the `values` array and that's our data.